test/test.js

```diff
@@ -5,16 +5,53 @@ import { test } from '../src/index.js';
 describe('test', function () {
   it('should detect as path template', function () {
     assert.isTrue(test('/{path}'));
+    // trailing slash is allowed
+    assert.isTrue(test('/{path}/'));
     assert.isTrue(test('/pets/{petId}'));
     assert.isTrue(test('/{entity}/me'));
     assert.isTrue(test('/books/{id}'));
     assert.isTrue(test('/a{test}'));
+    assert.isTrue(test('/{test}a'));
+    // parentheses are allowed
+    assert.isTrue(test('/range({x},{y})'));
+    // repeated parameter names are allowed
+    assert.isTrue(test('/range({x},{y})/secondRange({x},{y})'));
     assert.isTrue(test('/{entity}/{another-entity}/me'));
+    // NOTE: the following scenario should probably be considered invalid, query parameters are not allowed in path templates
     assert.isTrue(test('/pets?offset=0&limit=10'));
+    // -->
     assert.isTrue(test('/'));
+    // special characters in literal are allowed
+    assert.isTrue(test('/-/'));
+    assert.isTrue(test('/~/'));
+    // NOTE: the following scenario should probably be considered invalid, fragments are not allowed in path templates
+    assert.isTrue(test('/#baz'));
+    // NOTE: the following scenarios should probably be considered invalid, should we allow RFC 6570 operators in parameter names?
+    assert.isTrue(test('/{+baz}'));
+    assert.isTrue(test('/{;baz}'));
+    assert.isTrue(test('/{&baz}'));
+    assert.isTrue(test('/{.baz}'));
+    assert.isTrue(test('/{count*}'));
+    assert.isTrue(test('/{y,x}'), '/{y,x}');
+    // NOTE: the following scenarios should probably be considered invalid due to the use of special characters
```

Comment:
These characters are from https://datatracker.ietf.org/doc/html/rfc6570#section-1.5 and are perfectly fine percent unencoded as part of the [Path](https://datatracker.ietf.org/doc/html/rfc3986#section-3.3).

```abnf
gen-delims     =  ":" / "/" / "?" / "#" / "[" / "]" / "@"
sub-delims     =  "!" / "$" / "&" / "'" / "(" / ")"
                    /  "*" / "+" / "," / ";" / "="
```

Reply:
The point I was trying to make with this comment is those characters should not be allowed in a variable name. I agree with you they should be allowed in the path.

Reply:
@baywet IMHO there's a gotcha. `Parameter Object.name` does not have any constraint except that it must be string. If you try to constraint variable names which must match `Parameter Object.name`, you would automatically constraint the `Parameter Object.name` as well. That's why we're now only discussing that the only constraint is the prop application of percent encoding, see https://github.com/OAI/OpenAPI-Specification/issues/3256#issuecomment-2173808402.

Reply:
From the TDC call, same as the previous batch, doesn't fail, but doesn't do anything.

Reply:
ACK